src/Microsoft.DotNet.Wpf/src/PresentationFramework/Microsoft/Win32/FileDialog.cs

```diff
@@ -1110,26 +1109,8 @@ internal virtual bool PromptUserIfAppropriate(string fileName)
             // PromptUserIfAppropriate is OFN_FILEMUSTEXIST.
             if (GetOption(NativeMethods.OFN_FILEMUSTEXIST))
             {
-                try
-                {
-                    // File.Exists requires a full path, so we call GetFullPath on
-                    // the filename before checking if it exists.
-                    (new FileIOPermission(FileIOPermissionAccess.Read | FileIOPermissionAccess.PathDiscovery, fileName)).Assert();
-                    try
-                    {
-                        string tempPath = Path.GetFullPath(fileName);
-                        fileExists = File.Exists(tempPath);
-                    }
-                    finally
-                    {
-                        CodeAccessPermission.RevertAssert();
-                    }
-                }
-                // FileIOPermission constructor will throw on invalid paths.
-                catch (PathTooLongException)
```

Comment:
`PathTooLongException` should be retained

Reply:
`Path.GetFullpath()` can also throw `PathTooLongException`

See https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getfullpath?view=netframework-4.8#System_IO_Path_GetFullPath_System_String_